### PR TITLE
simplify and make it easier to read

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Refactoring.

## Motivation and Context

Reduce per method complexity and make it easier to read.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by honnix** [comment]
codacy doesn't understand custom assertion :(

**Comment by honnix** [comment]
@danielnorberg PTAL

**Comment by danielnorberg** [comment]
Dang, I thought codacy / PMD would recognice custom asserts. This rule has cause trouble before and it has at least never helped me find a case where I forgot to add an assert to my tests. I'd be fine with disabling it.

Alternatively we could switch back to returning bool.

File: styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java [hunk 1]
```diff
@@ -35,7 +35,6 @@ private ReplayEvents() {
     throw new UnsupportedOperationException();
   }
 
-  // TODO: fix NPath complexity
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java]: Codacy doesn't seem to think that the NPath complexity issue has been resolved

**Comment by danielnorberg** [comment]
We might need to refactor a bit more to get the complexity down.

File: styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java (hunk 1 above)
Comment (honnix) [reply styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java]: Weird. This method is not that complex now I think.

Comment (honnix) [reply styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java]: Now it has been resolved.

**Comment by honnix** [comment]
This is included in  #588

**Comment by danielnorberg** [comment]
Codacy still says “The method getBackfillRunState() has an NPath complexity of 500”

**Pull request closed without merging**